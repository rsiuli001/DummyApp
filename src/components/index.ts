export { default as ScreenList } from './ScreenList/ScreenList';
export { default as Buttons } from './Buttons/Buttons';
export { default as Banner } from './Banner/Banner';
