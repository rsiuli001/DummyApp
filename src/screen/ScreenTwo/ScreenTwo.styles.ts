import { StyleSheet } from 'react-native';
import * as Colors from '../../constants/colors';

const ScreenTwoStyles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: Colors.BLACK,
  },
  textContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'center',
    marginRight: 20,
    marginTop: 20,
    marginBottom: 10,
  },
  text: {
    fontWeight: '400',
    fontSize: 16,
    color: Colors.WHITE,
  },
});

export default ScreenTwoStyles;
