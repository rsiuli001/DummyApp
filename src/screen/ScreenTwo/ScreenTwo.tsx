import React from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';
import { Buttons, ScreenList } from '../../components';
import ScreenTwoStyles from './ScreenTwo.styles';
import * as ScreenName from '../../constants/screenName';
import * as Strings from '../../constants/strings';
import * as Colors from '../../constants/colors';

interface ScreenTwoProps {
  userName: String;
  navigation: any;
}

export const ScreenTwo: React.FC<ScreenTwoProps> = ({
  userName,
  navigation,
}): JSX.Element => {
  const onPress = (screenName: string) => {
    navigation.navigate(screenName);
  };

  return (
    <View style={ScreenTwoStyles.container}>
      <View style={ScreenTwoStyles.textContainer}>
        <Text style={ScreenTwoStyles.text}>
          {Strings.WELCOME} {userName}
        </Text>
      </View>

      <ScreenList
        text={`${Strings.NAVIGATE_TO} ${Strings.SCREEN_ONE}`}
        color={Colors.WHITE}
        onPress={onPress}
        screenName={ScreenName.SCREEN_ONE}
      />
      <ScreenList
        text={`${Strings.NAVIGATE_TO} ${Strings.SCREEN_TWO}`}
        color={Colors.WHITE}
        onPress={onPress}
        screenName={ScreenName.SCREEN_TWO}
      />
      <ScreenList
        text={`${Strings.NAVIGATE_TO} ${Strings.SCREEN_THREE}`}
        color={Colors.WHITE}
        onPress={onPress}
        screenName={ScreenName.SCREEN_THREE}
      />

      <Buttons></Buttons>
    </View>
  );
};

const mapStateToProps = (state: ScreenTwoProps) => ({
  userName: state.userName,
});

export default connect(mapStateToProps)(ScreenTwo);
