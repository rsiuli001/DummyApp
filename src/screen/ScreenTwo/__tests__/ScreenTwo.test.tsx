import React from 'react';
import { shallow } from 'enzyme';
import { ScreenTwo } from '../ScreenTwo';
import { View } from 'react-native';
import { Buttons, ScreenList } from '../../../components';

describe('Screen two Tests', () => {
  describe('rendering screen two', () => {
    const wrapper = shallow(<ScreenTwo userName="aa" navigation={{}} />);

    it('should render view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(View).length).toEqual(2);
    });

    it('should render Buttons component', () => {
      expect(wrapper.find(Buttons).length).toEqual(1);
    });

    it('should render ScreenList component', () => {
      expect(wrapper.find(ScreenList).length).toEqual(3);
    });
  });
});
