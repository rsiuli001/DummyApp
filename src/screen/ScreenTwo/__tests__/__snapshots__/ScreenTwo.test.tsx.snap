// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen two Tests rendering screen two should render view 1`] = `ShallowWrapper {}`;
